control/optimal.py

```diff
@@ -20,8 +20,6 @@
 from .exception import ControlNotImplemented
 from .timeresp import TimeResponseData
 
-__all__ = ['find_optimal_input']
-
```

Comment:
Should this have the "public" `*_constraint()` functions and the Class instead, @murrayrm?

Reply:
The `optimal` module is not currently imported in `__init__.py` and so you have to import the optimal control module explicitly (eg, `import control.optimal as opt`).  This is because those functions are still a bit new and non-standard => I felt like we should make users do a bit more work.

I don't know where this `__all__` line came from, but it is definitely wrong (the function doesn't exist).  Seems like we can either leave it out entirely (as proposed here) or put in all of the public functions and classes.  I guess the latter is mainly useful if you want to do `from control.optimal import *`.  I don't have a strong preference, but would be happy to hear from others if there is a general guideline that we should follow.

Reply:
No strong preference here either. Leaving it out entirely seems reasonable to me. `from foo import *`  is considered an anti-pattern by many. We do not prevent it, but getting the one private variable here as well as the external and internal imports is not that harmful either.